rust/lance/src/dataset.rs

```diff
@@ -878,9 +878,16 @@ impl Dataset {
 
         let remapping_index: UInt64Array = row_ids
             .iter()
-            .filter_map(|o| returned_row_ids.binary_search(o).ok().map(|pos| pos as u64))
+            .filter_map(|o| {
+                returned_row_ids
+                    .iter()
+                    .position(|id| id == o)
+                    .map(|pos| pos as u64)
+            })
             .collect();
 
+        debug_assert_eq!(remapping_index.len(), one_batch.num_rows());
```

Comment:
should we remove this?

Reply:
I'd like to keep it. It's a reminder of the invariant we need to keep in the function, and is ignored in release builds.